gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/MongoPortalPageContentRepository.java

```diff
@@ -0,0 +1,131 @@
+            logger.error("Failed to find portal page content by id", ex);
+            throw new TechnicalException("Failed to find portal page content by id", ex);
+        }
+    }
+
+    @Override
+    public PortalPageContent create(PortalPageContent item) throws TechnicalException {
+        try {
+            PortalPageContentMongo created = internalRepo.insert(map(item));
+            return map(created);
+        } catch (Exception ex) {
+            logger.error("Failed to create portal page content", ex);
+            throw new TechnicalException("Failed to create portal page content", ex);
+        }
+    }
+
+    @Override
+    public PortalPageContent update(PortalPageContent item) throws TechnicalException {
+        try {
+            PortalPageContentMongo saved = internalRepo.save(map(item));
+            return map(saved);
+        } catch (Exception ex) {
+            logger.error("Failed to update portal page content", ex);
+            throw new TechnicalException("Failed to update portal page content", ex);
+        }
+    }
+
+    @Override
+    public void delete(String id) throws TechnicalException {
+        try {
+            internalRepo.deleteById(id);
+        } catch (Exception ex) {
+            logger.error("Failed to delete portal page content [{}]", id, ex);
+            throw new TechnicalException("Failed to delete portal page content", ex);
+        }
+    }
+
+    @Override
+    public void deleteByType(PortalPageContent.Type type) throws TechnicalException {
+        try {
+            internalRepo.deleteByType(type);
+        } catch (Exception ex) {
+            logger.error("Failed to delete portal page contents by type", ex);
+            throw new TechnicalException("Failed to delete portal page contents by type", ex);
+        }
+    }
+
+    @Override
+    public java.util.Set<PortalPageContent> findAll() throws TechnicalException {
+        try {
+            java.util.List<PortalPageContentMongo> results = internalRepo.findAll();
+            return results.stream().map(this::map).collect(Collectors.toSet());
+        } catch (Exception ex) {
+            logger.error("Failed to find all portal page contents", ex);
+            throw new TechnicalException("Failed to find all portal page contents", ex);
+        }
+    }
+
+    private PortalPageContent map(PortalPageContentMongo mongo) {
+        PortalPageContent p = new PortalPageContent();
+        p.setId(mongo.getId());
+        p.setType(mongo.getType());
+        p.setConfiguration(mongo.getConfiguration());
+        p.setContent(mongo.getContent());
+        return p;
+    }
+
+    private PortalPageContentMongo map(PortalPageContent item) {
+        PortalPageContentMongo mongo = new PortalPageContentMongo();
+        mongo.setId(item.getId());
+        mongo.setType(item.getType());
+        mongo.setConfiguration(item.getConfiguration());
+        mongo.setContent(item.getContent());
+        return mongo;
+    }
```

Comment:
For the mapping in Mongo, we are using MapStruct.
A GraviteeMapper interface exists: `io/gravitee/repository/mongodb/management/mapper/GraviteeMapper.java`

you can use it in your class:

```
    @Autowired
    private GraviteeMapper mapper;
```